cmd/kubeadm/app/phases/controlplane/users_linux.go

```diff
@@ -0,0 +1,260 @@
+// +build linux
+
+package controlplane
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"strconv"
+	"strings"
+
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	"k8s.io/utils/exec"
+)
+
+var ()
+
+var UpdateFileOwnershipFn = func(filepath string, uid int64, gid int64, permissions uint32) error {
+	if err := os.Chown(filepath, int(uid), int(gid)); err != nil {
+		return err
+	}
+	if err := os.Chmod(filepath, fs.FileMode(permissions)); err != nil {
+		return err
+	}
+	return nil
+}
+
+func UpdateFileOwnership(filepath string, uid int64, gid int64, permissions uint32) error {
+	return UpdateFileOwnershipFn(filepath, uid, gid, permissions)
+}
+
+var UpdateDirecotryOwnershipFn = func(directorypath string, uid int64, gid int64, permissions string) error {
```

Comment:
This needs a comment and warning that this should not be called directly and is only for testing to override the function behavior for tests.

Also I hate that I have to do this and am open to suggestions on doing it another way if possible.

Reply:
similar comment to [1]

Reply:
typo in `UpdateDirecotryOwnershipFn`